OpenTelemetry (S3M) MS/Azure apps (#1797)
SumoLogic/sumologic-documentation: docs/integrations/microsoft-azure/opentelemetry/active-directory-json-opentelemetry.md

---
id: active-directory-json-opentelemetry
title: Active Directory JSON - OpenTelemetry Collector
sidebar_label: Active Directory JSON - OTel Collector
description: Learn about the Sumo Logic OpenTelemetry App for Active Directory JSON.
---

import useBaseUrl from '@docusaurus/useBaseUrl';
import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

> <img src={useBaseUrl('img/integrations/microsoft-azure/windows.png')} alt="Thumbnail icon" width="40"/> <img src={useBaseUrl('img/send-data/otel-color.svg')} alt="Thumbnail icon" width="45"/>

Reviewer: The image link is broken.

We recommend using the Active Directory JSON App in combination with the Windows JSON App.Active Directory logs are sent to Sumo Logic through OpenTelemetry [windowseventlogreceiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/windowseventlogreceiver).

<img src={useBaseUrl('https://https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Active-Directory-OpenTelemetry/Active-Directory-Schematics.png')} alt="Schematics" />

## Fields creation in Sumo Logic for Active Directory

Following are the [fields](https://help.sumologic.com/docs/manage/fields/) which will be created as part of Active Directory App install if not already present.

**`sumo.datasource`** - Has fixed value of **activeDirectory**

### Event Logs Used by Active Directory App

This section provides instructions for configuring log collection for Active Directory running on a non-Kubernetes environment for the Sumo Logic App for Active Directory.

#### Log Types

Standard Windows event channels include:
-   **Security**
-   **System**
-   **Application**

> ## Collection Configuration & App Installation

Reviewer: ```suggestion
## Collection Configuration and App Installation
```

> As part of the setting up the collection process and app installation, user can select the App from App Catalog and click on Install App. Please follow the steps below:

Reviewer: ```suggestion
As part of the setting up the collection process and app installation, you can select the App from App Catalog and click on Install App. Please follow the steps below:
```

### Step 1: Set up Collector

If you want to use an existing OpenTelemetry Collector then this step can be skipped by selecting the option of using an existing Collector.

If you want to create a new Collector please select **Add a new Collector** option.

This will generate a command which can be executed in the windows machine which needs to get monitored. Once executed it will install the Sumo Logic OpenTelemetry Collector agent.

> <img src={useBaseUrl('https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Active-Directory-OpenTelemetry/Active-Directory-Collector.png')} alt="Collector" />

Reviewer: The image link is broken when I view the page locally (by running yarn).

### Step 2: Configure Integration

In this step, we will be configuring the yaml required for Active Directory Collection.

You can add any custom fields which you want to tag along with the data ingested in sumo.

Click on the **Download YAML File** button to get the yaml file.

> <img src={useBaseUrl('https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Active-Directory-OpenTelemetry/Active-Directory-YAML.png')} alt="YAML" />

Reviewer: The image link is broken when I view the page locally (by running yarn).

### Step 3: Sending logs to Sumo

Once you have the yaml file downloaded in step 2, follow the steps below.

> 1.  Copy the yaml at *C:\ProgramData\Sumo Logic\OpenTelemetry Collector\config\conf.d*  folder in the machine which needs to be monitored.

Reviewer: ```suggestion
1.  Copy the yaml at `C:\ProgramData\Sumo Logic\OpenTelemetry Collector\config\conf.d`  folder in the machine which needs to be monitored.
```

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

<img src={useBaseUrl('img/integrations/microsoft-azure/windows.png')} alt="Thumbnail icon" width="40"/> <img src={useBaseUrl('img/send-data/otel-color.svg')} alt="Thumbnail icon" width="45"/>

The [Active Directory](https://learn.microsoft.com/en-us/windows-server/identity/ad-ds/get-started/virtual-dc/active-directory-domain-services-overview) JSON App helps you monitor your Windows Active Directory deployment by analyzing Active Directory logs in the JSON based event log format. The app includes predefined searches and dashboards that provide user activity into your environment for real-time analysis of overall usage.

We recommend using the Active Directory JSON App in combination with the Windows JSON App.Active Directory logs are sent to Sumo Logic through OpenTelemetry [windowseventlogreceiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/windowseventlogreceiver).

<img src={useBaseUrl('https://https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Active-Directory-OpenTelemetry/Active-Directory-Schematics.png')} alt="Schematics" />

> ## Fields creation in Sumo Logic for Active Directory

Reviewer 2: 1. Use relative paths - `/docs/manage/fields/`
2. I see only a single field under this heading.

Following are the [fields](https://help.sumologic.com/docs/manage/fields/) which will be created as part of Active Directory App install if not already present.

> **`sumo.datasource`** - has fixed value of activeDirectory

Reviewer 3: ```suggestion
**`sumo.datasource`** - Has fixed value of **activeDirectory**
```

### Event Logs Used by Active Directory App

This section provides instructions for configuring log collection for Active Directory running on a non-Kubernetes environment for the Sumo Logic App for Active Directory.

> #### Log Types

Reviewer 2: This could be the third heading.

Standard Windows event channels include:

> 

Reviewer 2: ```suggestion
```

-   **Security**
-   **System**
-   **Application**

> ## Collection Configuration & App installation

Reviewer 2: ```suggestion
## Collection Configuration & App Installation
```

> As part of the setting up the collection process and app installation user can select the App from App Catalog and click on Install App. Please follow the steps below:

Reviewer 3: ```suggestion
As part of the setting up the collection process and app installation, user can select the App from App Catalog and click on Install App. Please follow the steps below:
```

> As part of the setting up the collection process and app installation, user can select the App from App Catalog and click on Install App. Please follow the steps below:

Reviewer 2: ```suggestion
As part of setting up the collection process and app installation, you can select the App from App Catalog and click on Install App. Follow the steps below.
```

Reviewer 2: There is no "Install App" button anymore. You should update the installation steps. It's "Add Integration" button.

Author: @JainM6 Great catch!! Thank you so much.

Author: I checked with the S3M team. It looks like they're reverting back to **Install App** in the new UI.

<img width="497" alt="Screenshot 2023-03-25 at 9 59 20 AM" src="https://user-images.githubusercontent.com/56411016/227731459-8e7f662f-b22b-472e-9d16-d19c917264a6.png">

### Step 1: Set up Collector

If you want to use an existing OpenTelemetry Collector then this step can be skipped by selecting the option of using an existing Collector.

If you want to create a new Collector please select **Add a new Collector** option.

This will generate a command which can be executed in the windows machine which needs to get monitored. Once executed it will install the Sumo Logic OpenTelemetry Collector agent.

<img src={useBaseUrl('https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Active-Directory-OpenTelemetry/Active-Directory-Collector.png')} alt="Collector" />

> ### Step 2: Configure integration

Reviewer 2: ```suggestion
### Step 2: Configure Integration
```

> In this step we will be configuring the yaml required for Active Directory Collection.

Reviewer 2: ```suggestion
In this step, we will be configuring the yaml required for Active Directory Collection.
```

> You can add any additional fields which you want to tag as part of the data collected.

Reviewer 3: ```suggestion
You can add any custom fields which you want to tag along with the data ingested in sumo.
```

Click on the **Download YAML File** button to get the yaml file.

<img src={useBaseUrl('https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Active-Directory-OpenTelemetry/Active-Directory-YAML.png')} alt="YAML" />

### Step 3: Sending logs to Sumo

> Once you have the yaml file downloaded in step 2, please follow the below steps

Reviewer 2: ```suggestion
Once you have the yaml file downloaded in step 2, follow the below steps.
```

> 1.  Copy the yaml at `C:\ProgramData\Sumo Logic\OpenTelemetry Collector\config\conf.d` folder in the machine which needs to be monitored.

Reviewer 2: ```suggestion
1.  Copy the yaml at *C:\ProgramData\Sumo Logic\OpenTelemetry Collector\config\conf.d*  folder in the machine which needs to be monitored.
```

> 2.  restart the collector using the command `Restart-Service -Name`.

Reviewer 2: ```suggestion
2.  Restart the collector using the command `Restart-Service -Name`.
```

Reviewer 2: Not sure about the hyphen (-Name). I don't think there should be a space. @nmohantyctr , can you confirm this?

> After successful execution of the above command, Sumo will start receiving the data from your host machine.

Reviewer 2: ```suggestion
After the successful execution of the above command, Sumo Logic will start receiving the data from your host machine.
```

> This will install the app to your Sumo Logic Org. The app consists of Dashboards.

Reviewer 3: ```suggestion
Press **Next** . This will install the app to your Sumo Logic Org. The app consists of Dashboards.
```

details:"{\"Account For Which Logon Failed\":\"\",\"Network Information\":\"\",\"Failure Information\":\"\",\"Detailed Authentication Information\":\"\",\"Subject\":\"\",\"Process Information\":\"\",\"Logon Type\":\"3\",\"Additional Context\":\"\"}",
message:"An account failed to log on.",
event_id:"{\"qualifiers\":\"0\",\"id\":\"4625\"}",
level:"Information"
}
```

### Sample Query

This sample Query is from the Active Directory - Active Directory Service Activity > Top 10 Messages panel.

```sql
%"sumo.datasource"=activeDirectory
| json "event_id", "computer", "message" as event_id, host, msg_summary nodrop
| parse regex field=msg_summary "(?<msg_summary>.*\.*)" nodrop
| where host matches "*"
| count by msg_summary
| top 10 msg_summary by _count
```

## Viewing Active Directory Dashboards

### Active Directory Service Activity

-   The **Active Directory Service Activity** dashboard provides insights into overall active directory services like Category overtime, object creation, top 10 messages.

<img src={useBaseUrl('https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Active-Directory-OpenTelemetry/Active-Directory-Service-Activity.png')} alt="Service Activity" />

> ### Active Directory Service Failures:

Reviewer 2: ```suggestion
### Active Directory Service Failures
```
